components/gitpod-protocol/src/util/generate-workspace-id.ts

```diff
@@ -0,0 +1,454 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+export function generateWorkspaceID() {
```

Comment:
Same goes for TypeScript version: [`uuid` uses](https://github.com/uuidjs/uuid/blob/master/src/rng.js#L8) [randomFillSync](https://nodejs.org/docs/latest-v12.x/api/crypto.html#crypto_crypto_randomfillsync_buffer_offset_size).